tests/testsuite/message_format.rs

```diff
@@ -66,17 +66,23 @@ fn cargo_renders() {
 
     p.cargo("check --message-format json-render-diagnostics")
         .with_status(101)
-        .with_stdout(
-            "{\"reason\":\"compiler-artifact\",[..]\n\
-             {\"reason\":\"build-finished\",\"success\":false}",
-        )
-        .with_stderr_contains(
-            "\
-[CHECKING] bar [..]
-[CHECKING] foo [..]
-error[..]`main`[..]
-",
-        )
+        .with_stdout_data(str![[r#"
```

Comment:
This is something we want to avoid at this moment <https://github.com/rust-lang/cargo/issues/14039#issuecomment-2160827084>. Could you revert and annotate the function with `#[allow(deprecated)]`?

Reply:
addressed at commit a1f4d03